src/adapter/src/coord.rs

```diff
@@ -3459,6 +3492,22 @@ pub fn serve(
         );
         check_scheduling_policies_interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
 
+        let check_clusters_hydrated_interval = if read_only_controllers {
+            let dyncfgs = catalog.system_config().dyncfgs();
+            let interval = WITH_0DT_DEPLOYMENT_HYDRATION_CHECK_INTERVAL.get(dyncfgs);
+
+            let mut interval = tokio::time::interval(interval);
+            interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+            interval
+
+        } else {
+            // When not in read-only mode, we don't do hydration checks. But we
+            // still have to provide _some_ interval.
+            let mut interval = tokio::time::interval(Duration::from_millis(60 * 60 * 1000));
```

Comment:
I'm confused by this part! The duration selected is 1 hour, but why not use `Duration::from_hours`? Also, why is it one hour? Do we want the interval to tick every hour? If not, we should use `Duration::MAX`.

Reply:
Unfortunately, `from_hours` is unstable (same as `from_mins`). And using `Duration::MAX` will make the `tokio::interval::Interval` tick impl panic. So I picked a value that is large enough for it not to have an impact

But yes, it's annoying. All I really want is an interval that never triggers. I could do something more complicated here, like wrap the interval in an `async_stream`, and for the else case generate an `async_stream` that never yields. But that's more complicated code, and the type for that would be more complicated as well, so ... :man_shrugging:

Reply:
Looks like it will be fixed soon! https://github.com/tokio-rs/tokio/issues/6634